accord-core/src/test/java/accord/verify/ElleVerifier.java

```diff
@@ -0,0 +1,411 @@
+        PersistentArrayMap result = (PersistentArrayMap) Clj.check.invoke(Clj.elleListAppendOps, eventHistory);
+        Object isValid = result.get(Keys.valid);
+        if (isValid == Boolean.TRUE)
+            return;
+        if (isValid == Keys.unknown)
+        {
+            // Elle couldn't figure out if the history is bad or not... why?
+            Object anomalyTypes = result.get(Keys.anomalyTypes);
+            if (anomalyTypes != null)
+            {
+                ArraySeq seq = (ArraySeq) anomalyTypes;
+                if (!seq.isEmpty())
+                {
+                    boolean empty = false;
+                    for (Object type : seq)
+                    {
+                        if (type == Keys.emptyTransactionGraph)
+                        {
+                            empty = true;
+                            continue; // nothing to see here
+                        }
+                        throw new AssertionError("Unexpected anomaly type detected: " + type);
+                    }
+                    if (empty)
+                        return; // all good
+                }
+            }
+        }
+        throw new HistoryViolation(-1, "Violation detected: " + result);
+    }
+
+    private static abstract class Action extends java.util.AbstractList<Object> implements RandomAccess
+    {
+        enum Type
+        {
+            append, r;
+
+            final Keyword keyword;
+
+            Type()
+            {
+                keyword = RT.keyword(null, name());
+            }
+        }
+        private final Action.Type type;
+        private final int key;
+        private final Object value;
+
+        protected Action(Action.Type type, int key, Object value)
+        {
+            this.type = type;
+            this.key = key;
+            this.value = value;
+        }
+
+        @Override
+        public Object get(int index)
+        {
+            switch (index)
+            {
+                case 0:  return type.keyword;
+                case 1:  return key;
+                case 2:  return value;
+                default: throw new IndexOutOfBoundsException();
+            }
+        }
+
+        @Override
+        public int size()
+        {
+            return 3;
+        }
+    }
+
+    private static class Read extends Action
+    {
+        protected Read(int key, int[] seq)
+        {
+            // TODO (optimization): rather than vector of boxed int, can we use the interfaces so we can stay primitive array?
+            super(Type.r, key, PersistentVector.create(IntStream.of(seq).mapToObj(Integer::valueOf).collect(Collectors.toList())));
+        }
+    }
+
+    private static class Append extends Action
+    {
+        protected Append(int key, int value)
+        {
+            super(Type.append, key, value);
+        }
+    }
+
+    private static class Event extends ObjectPersistentMap
+    {
+        enum Type
+        {
+            ok, fail; // invoke, info
```

Comment:
To increase coverage, isn't `info` also required as a possible response type ([as per docs](https://github.com/jepsen-io/elle/blob/main/src/elle/list_append.clj#L582)), where we (on client side) do not know, if the transaction has been applied or if it failed?
I'm asking because in the [clojure based test](https://github.com/datastax/accordclient/blob/main/src/accordclient/core.clj#L85-L93) I've used for the algorithm verification I've mapped:
- `UnavailableException` and `NoHostAvailableException` to `:fail`
- `ReadTimeoutException`, `WriteTimeoutException` and `OperationTimedOutException` to `:info`

Wanted to make sure, that I'm not misinterpreting response types and `info` has been skipped deliberately.

Reply:
I think we probably also want to issue `:invoke` with the start time as well?

Reply:
I'm very surprised that the `:invoke` tag is ignored, as it seems impossible to distinguish between some valid and invalid traces without knowing whether two operations overlapped their execution or not.

Reply:
hmmm maybe I was confused reading the code... thought invoke is what we sent and info was random stuff like "added network partition"... ill try to look closer

Reply:
reading the code I see `dirty-update-cases` uses `:info` so will look to add that.  I did see `verify-unique-appends` check for `invoke`, this looks to make sure only a single `:invoke` wrote `X`... I don't see any other usage...

ill try adding both; this will add more memory but is more "correct"... still not sure that either really provides value for Elle, but I don't know Clojure well so maybe im missing something.

Reply:
The comments linked above suggest invoke isn’t used; I’m just very surprised by this if it’s true, as I can’t see how you would detect some invalid traces.

Reply:
joys... `invoke` means I must change the process ids... they don't allow concurrent process... so history#pair-index-fold fails with 

```
{:type :jepsen.history/double-invoke,
 :op
 {:process 0,
  :type :invoke,
  :f nil,
  :value [[:r 4]],
  :index 1,
  :time 60},
 :running
 {:process 0,
  :type :invoke,
  :f nil,
  :value [[:r 6]],
  :index 0,
  :time 3}}
```

Reply:
@belliottsmith , @kijanowski pushed a change to add `:invoke` and did see that it is used (it gets mad if you have concurrent invoke from the same `process`... so had to make every `process` our `start`; was planning to make it client but that would break the model)